vertical-pod-autoscaler/pkg/admission-controller/config.go

```diff
@@ -31,14 +32,43 @@ const (
 	webhookConfigName = "vpa-webhook-config"
 )
 
-func configTLS(serverCert, serverKey []byte) *tls.Config {
+func configTLS(serverCert, serverKey []byte, minTlsVersion, ciphers string) *tls.Config {
+	var tlsVersion uint16
+	var ciphersuites []uint16
+	reverseCipherMap := make(map[string]uint16)
 	sCert, err := tls.X509KeyPair(serverCert, serverKey)
 	if err != nil {
 		klog.Fatal(err)
 	}
+
+	for _, c := range tls.CipherSuites() {
+		reverseCipherMap[c.Name] = c.ID
+	}
+	for _, c := range strings.Split(strings.ReplaceAll(ciphers, ",", ":"), ":") {
+		cipher, ok := reverseCipherMap[c]
+		if ok {
+			ciphersuites = append(ciphersuites, cipher)
+		}
+	}
+	if len(ciphersuites) == 0 {
+		ciphersuites = nil
+	}
+
+	switch minTlsVersion {
+	case "tls1_0":
+		tlsVersion = tls.VersionTLS10
+	case "tls1_1":
+		tlsVersion = tls.VersionTLS11
+	case "tls1_3":
+		tlsVersion = tls.VersionTLS13
+	default:
+		tlsVersion = tls.VersionTLS12
```

Comment:
This is somewhat dangerous, IMHO. This code allows for arbitrary strings to be set for `--min-tls-version` and any typo will not result in an error, but in TLS1.2 being set. Users will never realize they're not getting what they intended. For example, if I'm setting `--min-tls-version=1.3`, admission-controller will silently configure TLS1.2. I think we should make it explicit to the user that they entered a value that is unknown and fail to start.

Reply:
updated code to default to `tls1_2` and only accept `tls1_2` and `tls1_3`.  If a different value is given, we now throw a fatal exception.  Error now looks like
```
F0318 15:31:04.356258       1 config.go:66] Unable to determine value for --min-tls-version (tls1_1), must be either tls1_2 or tls1_3
```